infrastructure/shared/main.tf

```diff
@@ -0,0 +1,79 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.112.0"
+    }
+  }
+
+  # backend "azurerm" {
+  #   use_azuread_auth = true
+  # }
+}
+
+
+locals {
+  environment = lower(var.environment)
+  metadata = {
+    solution    = "shared"
+    environment = local.environment
+    instance    = var.instance
+  }
+}
+
+resource "azurerm_resource_group" "authorization" {
+  name     = "rg-${local.metadata.solution}-${local.metadata.environment}-${local.metadata.instance}"
+  location = "norwayeast"
+}
+
+resource "azurerm_user_assigned_identity" "shared" {
```

Comment:
Hva er en "shared" user assigned identity?

Reply:
den blir ikke brukt helt enda, men tjenester som service bus osv. kommer til å få tilknyttet denne identiteten får å hente krypteringsnøkler fra key vault

Reply:
Burde de ikke ha hver sin identitet heller?

Reply:
Denne identitet er primært for tjenester som skal lese krypteringsnøkler fra key vault. IAM ellers bruker vi system assign identity.

Reply:
Skjoente det - men hvorfor ikke ha en identity per ting som skal lese krypteringsnokler? Da kan disse identityene lages i modulen for servicebus/db/whatever, i stedet for her.

Reply:
du kan ikke lage en service bus med encryption enabled og bruke System Assigned identity. Fordi system identiten har ikke rettigheter til å lese nøkler når du oppretter den sammen med ressursen. 

1. Lag identitet
2. Gi identitet rettighet til key vault
3. Opprett tjeneste med encryption enabled og at den bruker identiteten som har rettighet til å lese nøkler

Reply:
vi kan sikkert pakketere at alle modulene lager sin egen user assigned identity, tenkte det var greit å slippe lage x antall med samme hensikt

Reply:
Foeler det er ryddigere at dem lager sin egen. Ihvertfall hvis moduler skal fjaernes, saa rydder de bort etter seg.

Reply:
det kan vi ordne, da bare fjerner jeg ressursen fra main.tf